src/MetalPerformanceShaders/MPSKernel.cs

```diff
@@ -20,6 +20,18 @@ public static bool Supports (IMTLDevice device)
 			return MPSSupportsMTLDevice (device == null ? IntPtr.Zero : device.Handle);
 		}
 
+		[DllImport (Constants.MetalPerformanceShadersLibrary)]
+		[Introduced (PlatformName.MacCatalyst, 13, 0)]
+		[TV (12,2), Mac (10,14,4), iOS (12,2)]
```

Comment:
It did not show in previous API diff, e.g. for Xcode 10.3 -> https://github.com/xamarin/xamarin-macios/wiki/xcode103-Bindings-Status

So it _might_ be available in earlier OS (it happens) but even more commonly it's a mistake header (soft _lie_) from Apple. Running the tests on older devices or simulator (if available for those versions) is the only way to be sure if the correct availability. IOW don't trust headers if they do not match the current SDK version :)

Reply:
Yeah, it's odd. There are a lot of APIs, especially in MPSNeuralNetwork, that get this iOS 12.2 designation.

So what should I do for the attributes? Manually test? Or is there build step configuration for that version?

Also, Apple definitely didn't talk about these APIs until iOS 13, so I don't see any harm in fibbing a little and marking these iOS 13.

Reply:
The easy way is to run the introspection tests on iOS 12.2 and see if it can find the selectors.

In doubt (e.g. does not work for protocols) when use iOS 13.0 (the current SDK version) and we can refine this later (it's not a breaking change) and, like you said, it won't cause much of a problem for most developers :)